api/client/src/main/java/org/projectnessie/client/http/impl/HttpUtils.java

```diff
@@ -64,4 +68,27 @@ public static void applyHeaders(HttpHeaders headers, URLConnection con) {
       }
     }
   }
+
+  /**
+   * Parse query parameters from URI. This method cannot handle multiple values for the same
+   * parameter.
+   *
+   * @param query They query string to parse
+   * @return map of query parameters
+   */
+  public static Map<String, String> parseQueryString(String query)
+      throws UnsupportedEncodingException {
+    if (query == null) {
+      throw new IllegalArgumentException("Missing query string");
+    }
+    Map<String, String> params = new HashMap<>();
+    String[] pairs = query.split("&");
+    for (String pair : pairs) {
+      int idx = pair.indexOf("=");
+      String name = URLDecoder.decode(pair.substring(0, idx), "UTF-8");
```

Comment:
`StandardCharsets.UTF_8`?

Reply:
Unfortunately, that method is only available since Java 10, and here we must ensure java 8 compatibility. But indeed I should not throw `UnsupportedEncodingException` here, I will change that.